From 864fe1f93b6a46a265430e8968d03a4cb57e93c5 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Thu, 15 Jun 2023 19:49:59 +0200
Subject: [PATCH] gmscompat: Spoof model, product and fingerprint with cust  
 prop

gmscompat: Spoof model, product and fingerprint with cust
 prop

Allows you to change the product, device,  model and the fingerprint during the Safetynet certification

use severals properties : persist.sys.phh.safetyspoof, persist.sys.phh.safetyspoof.model, etc...

Change-Id: If886a953d2d0165ed1f37b8e12dbc04d338536a9
---
 .../internal/util/PropImitationHooks.java     | 22 ++++++++++++++-----
 1 file changed, 17 insertions(+), 5 deletions(-)

diff --git a/core/java/com/android/internal/util/PropImitationHooks.java b/core/java/com/android/internal/util/PropImitationHooks.java
index c844ecf94038..78135d6be7bd 100644
--- a/core/java/com/android/internal/util/PropImitationHooks.java
+++ b/core/java/com/android/internal/util/PropImitationHooks.java
@@ -71,6 +71,18 @@ public class PropImitationHooks {
         "PIXEL_2021_MIDYEAR_EXPERIENCE"
     );
 
+    private static final boolean SAFETYSPOOF_ENABLE =
+            android.os.SystemProperties.getBoolean("persist.sys.phh.safetyspoof", false);
+    private static final String SAFETYSPOOF_MODEL =
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.model", "Pixel XL");
+    private static final String SAFETYSPOOF_DEVICE =
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.device", "marlin");
+    private static final String SAFETYSPOOF_PRODUCT =
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.product", "marlin");
+    private static final String SAFETYSPOOF_FINGERPRINT=
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.fingerprint", "google/marlin/marlin:7.1.2/NJH47F/4146041:user/release-keys");
+
+
     private static volatile boolean sIsGms = false;
     private static volatile boolean sIsFinsky = false;
     private static volatile boolean sIsPhotos = false;
@@ -92,12 +104,12 @@ public class PropImitationHooks {
          * Set stock fingerprint for ARCore
          * Set Pixel XL for Google Photos and Snapchat
          */
-        if (sCertifiedProps.length == 4 && sIsGms) {
+        if (sCertifiedProps.length == 4 && sIsGms && SAFETYSPOOF_ENABLE) {
             dlog("Spoofing build for GMS");
-            setPropValue("DEVICE", sCertifiedProps[0]);
-            setPropValue("PRODUCT", sCertifiedProps[1]);
-            setPropValue("MODEL", sCertifiedProps[2]);
-            setPropValue("FINGERPRINT", sCertifiedProps[3]);
+            setPropValue("DEVICE", SAFETYSPOOF_DEVICE);
+            setPropValue("PRODUCT", SAFETYSPOOF_PRODUCT);
+            setPropValue("MODEL", SAFETYSPOOF_MODEL);
+            setPropValue("FINGERPRINT", SAFETYSPOOF_FINGERPRINT);
         } else if (!sStockFp.isEmpty() && packageName.equals(PACKAGE_ARCORE)) {
             dlog("Setting stock fingerprint for: " + packageName);
             setPropValue("FINGERPRINT", sStockFp);
-- 
2.25.1

